### DISCUSSION TITLE
Add the upload modal to Medusa UI components

### DISCUSSION THREAD

**Discussion by mortenengel** [medusajs/medusa#4960] [Feature Requests]
First off, thumbs up on the new admin extensibility, and the Medusa UI components that has been added. Great work!

I would like to request that one of your current components from the admin UI is exported - preferably though the UI, but alternatively just from the current version in the admin-ui project.

We have created a custom strategy for importing customers and gift-cards from CSV files, and would very much like to use the same import modal that is used for the products and pricing import;
https://github.com/medusajs/medusa/blob/develop/packages/admin-ui/ui/src/components/organisms/upload-modal/index.tsx

Being able to use it exactly the same way as it's used from the price import today, would be perfect:
``` 
<UploadModal
      type="prices"
      fileTitle="Price List prices"
      description1Text="Upload a CSV file with variants and prices to update your price list. Note that any existing prices will be deleted."
      description2Title="Unsure about how to arrange your list?"
      description2Text="Download the template file below and update your prices"
      status={status}
      progress={progress}
      canImport={isPreprocessed}
      onSubmit={onSubmit}
      onClose={onClose}
      summary={getSummary()}
      onFileRemove={onFileRemove}
      processUpload={processUpload}
      onDownloadTemplate={downloadPricingImportCSVTemplate}
    />
```

Note, that in order to use the polling and notification context, we would need the useNotification and usePolling to be exported as well (which might make it hard to move it to the UI components?) https://github.com/medusajs/medusa/blob/develop/packages/admin-ui/ui/src/domain/pricing/batch-job/import.tsx

**Comment by hamudeshahin** [comment]
+1

**Comment by zoloogg** [comment]
+1
